tests/validation/tests/v3_api/test_deploy_clusters.py

```diff
@@ -51,20 +60,99 @@ def test_deploy_rke():
     create_config_file(env_details)
 
 
 @if_not_auto_deploy_eks
 def test_deploy_eks():
     print("Deploying EKS Clusters")
+    global env_details
+    errors = []
+    k8s_versions = [EKS_K8S_VERSIONS[0], EKS_K8S_VERSIONS[-1]]
+
+    for version in k8s_versions:
+        if env_details != "env.CLUSTER_NAMES='":
+            env_details += ","
+        try:
+            print("Deploying EKS Cluster using kubernetes version {}".format(
+                version))
+            client, cluster = create_and_validate_eks_cluster(version)
+            env_details += cluster.name
+        except Exception as e:
+            errors.append(e)
+
+    assert not errors
 
 
 @if_not_auto_deploy_gke
 def test_deploy_gke():
     print("Deploying GKE Clusters")
+    global env_details
+    errors = []
+
+    gke_versions, creds = get_gke_version_credentials(multiple_versions=True)
+
+    for i, version in enumerate(gke_versions, start=1):
+        c_name = "test-auto-gke-{}".format(i)
+        if env_details != "env.CLUSTER_NAMES='":
+            env_details += ","
+        try:
+            print("Deploying GKE Cluster using kubernetes version {}".format(
+                version))
+            client, cluster = create_and_validate_gke_cluster(c_name,
+                                                              version, creds)
+            env_details += cluster.name
+        except Exception as e:
+            errors.append(e)
+
+    assert not errors
 
 
 @if_not_auto_deploy_aks
 def test_deploy_aks():
     print("Deploying AKS Clusters")
+    global env_details
+    errors = []
+
+    aks_versions = get_aks_version(multiple_versions=True)
+
+    for version in aks_versions:
+        if env_details != "env.CLUSTER_NAMES='":
+            env_details += ","
+        try:
+            print("Deploying AKS Cluster using kubernetes version {}".format(
+                version))
+            client, cluster = create_and_validate_aks_cluster(version)
+            env_details += cluster.name
+        except Exception as e:
+            errors.append(e)
+
+    assert not errors
+
+
+@pytest.fixture(scope='module', autouse="True")
+def set_data(request):
+    print("In set_data function")
+    if KDM_BRANCH != "":
+        print("Updating KDM to use {}/tree/{}".format(KDM_URL, KDM_BRANCH))
+        header = {'Authorization': 'Bearer ' + ADMIN_TOKEN}
+        kdm_url = CATTLE_API_URL + "/settings/rke-metadata-config"
+        r = requests.put(kdm_url, verify=False, headers=header, json={
+            "name": "rke-metadata-config",
+            "value": json.dumps({
+                "branch": KDM_BRANCH,
+                "refresh-interval-minutes": "1440",
+                "url": KDM_URL
+            })
+        })
+        assert r.ok
+
+        # Refresh Kubernetes Metadata
+        kdm_refresh_url = CATTLE_API_URL + "/kontainerdrivers?action=refresh"
+        response = requests.post(kdm_refresh_url, verify=False, headers=header)
+        assert response.ok
```

Comment:
Can we obtain response.content after the post request and add asserts to check for the right values ?
json.loads(response.content) can be used if required.

Reply:
This value has `None` in response.content. I think for this one just checking that the response is 200 is enough since in the UI flow just a button click is all it is and it doesn't really give any "success" message. Let me know your thoughts!

Reply:
ok